osh/cmd_exec.py

```diff
@@ -379,6 +381,24 @@ def _RunBuiltin(self, builtin_id, argv, span_id):
     elif builtin_id == builtin_e.REPR:
       status = builtin.Repr(argv, self.mem)
 
+    elif builtin_id == builtin_e.BUILTIN:
+      if not argv:
+          status = 0
+      else:
+          to_run = builtin.Resolve(argv[0])
+          # check special builtins now, this is a strange thing to ask
+          # since they can't be overwritten but it is valid
+          if to_run == builtin_e.NONE:
+              to_run = builtin.ResolveSpecial(argv[0])
+          if to_run == builtin_e.NONE:
+              print("{}: {}builtin: {}: not a shell builtin".format(
```

Comment:
Please use `util.error()` here, following some of the examples in `osh/builtin.py`.  The error handling probably needs another pass, but having them all go through that function will make it easier to change at once.

Then you don't need to worry about the $0 issue... we can change that all at once if necessary.

I probably do want to change the interactive errors vs. batch errors as you've done, but it should be done globally.

Reply:
I can look at implementing this in `util.error`, I think having line numbers in error messages is easy and makes scripting much friendlier for the user.

Reply:
I'm not sure this can be done from `util`, those are all static methods and don't have information about the runtime. We could move `error` into somewhere osh-specific where it has access to the running interpreter, but I imagine that would require a pretty substantial rewrite.